tool/tsh/common/db.go

```diff
@@ -1039,25 +1048,37 @@ func listDatabasesByPrefix(ctx context.Context, tc *client.TeleportClient, prefi
 // a given additional predicate expression. If the teleport client already
 // has a predicate expression, the predicates are combined with a logical AND.
 func listDatabasesWithPredicate(ctx context.Context, tc *client.TeleportClient, predicate string) (types.Databases, error) {
-	if predicate == "" {
-		predicate = tc.PredicateExpression
-	} else if tc.PredicateExpression != "" {
-		predicate = fmt.Sprintf("(%v) && (%v)", predicate, tc.PredicateExpression)
-	}
 	var databases []types.Database
 	err := client.RetryWithRelogin(ctx, tc, func() error {
 		var err error
 		databases, err = tc.ListDatabases(ctx, &proto.ListResourcesRequest{
 			Namespace:           tc.Namespace,
 			ResourceType:        types.KindDatabaseServer,
-			PredicateExpression: predicate,
+			PredicateExpression: combinePredicateExpressions(predicate, tc.PredicateExpression),
 			Labels:              tc.Labels,
 		})
 		return trace.Wrap(err)
 	})
 	return databases, trace.Wrap(err)
 }
 
+// combinePredicateExpressions combines two predicate expressions into one
+// expression as a conjunction (logical AND) of the expressions.
+func combinePredicateExpressions(a, b string) string {
```

Comment:
I am surprised we don't have util to build these already. I think I only see one place using `vulcand/predicate/builder`.

Reply:
oh, lol `tsh` depends on lib/services which depends on vulcand. I guess it wouldn't add anything then 🤷‍♂️

Reply:
i played around with `vulcand/predicate/builder`, its not well written. Your version is a lot easier =D

Reply:
thanks for checking it out